### PR TITLE
[GPU] Set offsets of variadic split for static

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Details:
 - *Set offset of split at CreateCommonSplitOp for static shaped layer*

### Tickets:
 - *114145*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/crop.cpp [hunk 1]
```diff
@@ -134,6 +134,7 @@ std::vector<layout> crop_inst::calc_output_layouts(const crop_node& /*node*/, co
         }
 
         p_param->input_offsets[0] = ov::intel_gpu::tensor_from_dims(startOffset, 0);
+        const_cast<crop*>(desc.get())->offsets = p_param->input_offsets[0];
```
[the remark this answers is not in the document]
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/graph/crop.cpp:137]: The 'offsets' in cldnn::crop structure is needed but, impl_param is kernel_impl_params type and can't access it directly.
"desc->offsets = p_param->input_offsets[0]" looks possible but this has permissive error in compiling.

[the remark this answers is not in the document]
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/graph/crop.cpp]: This code is deleted and CreateCommonSplitOp step inserts calculated offset.

**Comment by yeonbok** [comment]
Please add a functional test.

File: src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/dynamic_model_static_split_layer.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        ElementType, // Network precision
+        TargetDevice, // Device name
+        std::map<std::string, std::string> // Additional network configuration
+> DynamicModelStaticSplitLayerGPUTestParamsSet;
+
+const std::vector<ElementType> netPrecisions = {
+    ElementType::f16,
+    // ElementType::f32,
+    // ElementType::i32,
+    // ElementType::i64,
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/dynamic_model_static_split_layer.cpp:31]: Please remove unused comments

File: src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/dynamic_model_static_split_layer.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
+// Copyright (C) 2018-2023 Intel Corporation
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/dynamic_model_static_split_layer.cpp:1]: nit:2023

File: src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/dynamic_model_static_split_layer.cpp (hunk 2 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/dynamic_model_static_split_layer.cpp]: done

File: src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/dynamic_model_static_split_layer.cpp (hunk 3 above)
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/dynamic_model_static_split_layer.cpp]: done

**Review by yeonbok — APPROVED** [review]

**Review by andrew-k-park — APPROVED** [review]

**Pull request merged**